src/Nodes1DProvisioner.cpp

```diff
@@ -71,6 +79,86 @@ void Nodes1DProvisioner::buildNodes() {
     }
 
     buildConnectivityMatrices();
+    buildFaceMask();
+}
+
+/**
+ * Build volume to surface maps.
+ */
+void Nodes1DProvisioner::buildMaps() {
+    firstIndex ii;
+    secondIndex jj;
+
+    index_matrix_type nodeIds(NumLocalPoints, NumElements);
+
+    // Set up reference to the objects we need to interact with.
+    SparseMatrixConverter & matConverter = *MatrixConverter;
+    
+    index_vector_type & Fmsk = *Fmask;
+    index_vector_type & vmM = *vmapM;
+    index_vector_type & vmP = *vmapP;
+
+    index_matrix_type & E2E = *EToE;
+    index_matrix_type & E2F = *EToF;
+    matrix_type & xmat = *xGrid;
+
+    matrix_type xmatTrans(NumElements, NumLocalPoints);
+    xmatTrans = xmat(jj,ii);
+
+    double * x = new double[NumElements*NumLocalPoints];
+    matConverter.fullToPodArray(xmatTrans, x);
+
+    // Assemble global volume node numbering.
+    nodeIds = ii + NumLocalPoints*jj;
+
+    vmM = 0*ii;
+    vmP = 0*ii;
+
+    index_type count=0;
+    for (index_type k = 0; k < NumElements; k++) {
+        for( index_type f = 0; f < NumFaces; f++ ) {
+            vmM(count) = nodeIds(Fmsk(f), k);
+            count++;
+        }
+    }
+
+    count = 0;
+    for (index_type k1=0; k1 < NumElements; k1++) {
+        for (index_type f1=0; f1 < NumFaces; f1++) {
+            index_type k2 = E2E(k1, f1);
+            index_type f2 = E2F(k1, f1);
+
+            index_type vidM = vmM(k1*NumFaces + f1);
+            index_type vidP = vmM(k2*NumFaces + f2);
+
+            real_type x1 = x[vidM];
+            real_type x2 = x[vidP];
+
+            real_type dist = sqrt(pow(x1 - x2, 2));
```

Comment:
Maybe change this to:
x1 -= x2;
real_type dist = sqrt(x1 * x1);

Reply:
hm, I don't really like treating x1 as a temp, it kinda throws away the "mathematicalness" of the current notation. What if we do sqrt((x1-x2)*(x1-x2)) ?